feat: Route - Controller Command

rolsonquadras opened the pull request:
- Add command method for Route registration for different bindings
- The command uses route client

part of #989 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

sudeshrshetty commented on pkg/controller/command/route/command.go:
```diff
@@ -0,0 +1,92 @@
+	InvalidRequestErrorCode = command.Code(iota + command.ROUTE)
+
+	// ResponseWriteErrorCode for response write error
+	ResponseWriteErrorCode
```
You shouldn't throw error response write, it **may** go in infinite loop. Since error handling again tries to write it to on same writer/response.

rolsonquadras replied:
[Removed](https://github.com/hyperledger/aries-framework-go/compare/e3d2bf8f0eff2b46016f1ba077db6cb3aefac866..2f1b5db8a74a7b7659c94b4d4c346b41e71d5c51#diff-d08f92ab0cfb3a36ef618e710ffc3fedL27)

sudeshrshetty commented on pkg/controller/command/route/command.go:
```diff
@@ -0,0 +1,92 @@
+func writeResponse(rw io.Writer, v interface{}) command.Error {
+	if err := json.NewEncoder(rw).Encode(v); err != nil {
+		logger.Errorf("Unable to send error response, %s", err)
+		return command.NewExecuteError(ResponseWriteErrorCode, err)
```
Shouldn't return error.

rolsonquadras replied:
[Refactored](https://github.com/hyperledger/aries-framework-go/compare/e3d2bf8f0eff2b46016f1ba077db6cb3aefac866..2f1b5db8a74a7b7659c94b4d4c346b41e71d5c51#diff-d08f92ab0cfb3a36ef618e710ffc3fedL88)

sudeshrshetty commented on pkg/controller/command/route/command.go:
```diff
@@ -0,0 +1,92 @@
+}
+
+// writeResponse writes interface value to response
+// TODO https://github.com/hyperledger/aries-framework-go/issues/1176 - Move these to common
```
Tried to move this part of separate packages, looked ugly :)
this function is nothing but a json encoder logic. For the sake of improving code coverage this one is moved to separate function in other commands

rolsonquadras replied:
Removed this TODO.

rolsonquadras committed:
feat: Route - Controller Command

- Add command method for Route registration for different bindings
- The command uses route client

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>
